src/BlockTypes/CollectionFilters.php

```diff
@@ -0,0 +1,179 @@
+	 *
+	 * @return null This block has no frontend script.
+	 */
+	protected function get_block_type_script( $key = null ) {
+		return null;
+	}
+
+	/**
+	 * Initialize this block type.
+	 *
+	 * - Hook into WP lifecycle.
+	 * - Register the block with WordPress.
+	 */
+	protected function initialize() {
+		parent::initialize();
+		add_action( 'render_block_context', array( $this, 'modify_inner_blocks_context' ), 10, 3 );
+	}
+
+	/**
+	 * Extra data passed through from server to client for block.
+	 *
+	 * @param array $attributes  Any attributes that currently are available from the block.
+	 *                           Note, this will be empty in the editor context when the block is
+	 *                           not in the post content on editor load.
+	 */
+	protected function enqueue_data( array $attributes = [] ) {
+		parent::enqueue_data( $attributes );
+
+		if ( ! is_admin() ) {
+			/**
+			 * At this point, WP starts rendering the Collection Filters block,
+			 * we can safely nuke the current response.
+			 */
+			$this->current_response = null;
+		}
+	}
+
+	/**
+	 * Modify the context of inner blocks.
+	 *
+	 * @param array    $context The block context.
+	 * @param array    $parsed_block The parsed block.
+	 * @param WP_Block $parent_block The parent block.
+	 * @return array
+	 */
+	public function modify_inner_blocks_context( $context, $parsed_block, $parent_block ) {
+		if (
+			is_admin() ||
+			! is_a( $parent_block, 'WP_Block' ) ||
+			"woocommerce/{$this->block_name}" !== $parent_block->name ||
+			empty( $parent_block->inner_blocks )
+		) {
+			return $context;
+		}
+
+		if ( ! isset( $this->current_response ) ) {
+			$this->current_response = $this->get_aggregated_collection_data( $parent_block );
+		}
+
+		if (
+			! empty(
+				$this->current_response ||
+				isset( $parsed_block['blockName'] ) ||
+				in_array( $parsed_block['blockName'], $this->collection_data_params_mapping, true )
+			) ) {
+			$context['collectionData'] = $this->current_response;
+		}
+
+		return $context;
+	}
+
+	/**
+	 * Get the aggregated collection data from the API.
+	 * Loop through inner blocks and build a query string to pass to the API.
+	 *
+	 * @param WP_Block $block The block instance.
+	 * @return array
+	 */
+	protected function get_aggregated_collection_data( $block ) {
+		$collection_data_params = array();
+		$inner_blocks           = array();
+
+		do {
+			$inner_blocks = array_merge(
+				$this->get_inner_blocks_recursive( $block->inner_blocks->current() ),
+				$inner_blocks
+			);
+			$block->inner_blocks->next();
+		} while ( $block->inner_blocks->valid() );
+
+		foreach ( $this->collection_data_params_mapping as $key => $block_name ) {
+			$collection_data_params[ $key ] = ( in_array( $block_name, $inner_blocks, true ) );
+		}
+
+		if ( empty( array_filter( $collection_data_params ) ) ) {
+			return array();
+		}
+
+		$response = Package::container()->get( Hydration::class )->get_rest_api_response_data(
```

Comment:
I'm a bit confused by this call. Who are all these classes? Not a problem with the code, I'm just wondering.

Reply:
They are referent at the top

```
use Automattic\WooCommerce\Blocks\Package;
use Automattic\WooCommerce\Blocks\Domain\Services\Hydration;
```

You can read more about them [here](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/Package.php) and [here](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/src/Domain/Services/Hydration.php).

Reply:
Because we have a container and already registered the `Hydration` class, I use it from the container instead of intializing it manually.